pkg/cmd/pulumi/convert_test.go

```diff
@@ -152,3 +154,50 @@ func TestProjectNameOverrides(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Contains(t, string(yamlBytes), "name: "+name)
 }
+
+// Tests that test for when args are passed to an imported resource that need
+// to be aliased are called with the appropriate alias, see (pulumi/pulumi#9664)
+func TestAliasedImportsWorkInArgs(t *testing.T) {
+	t.Parallel()
+
+	// Arrange.
+	outDir := t.TempDir()
+	cwd, err := filepath.Abs("convert_aliased_imports")
+	assert.NoError(t, err)
+
+	// Shell out pulumi plugin install pulumi-resource-aws and awsx, which is needed for the test
+	cmd := exec.Command("pulumi", "plugin", "install", "resource", "aws")
+	err = cmd.Run()
+	if err != nil {
+		log.Fatalf("Command execution failed: %v", err)
+	}
+	cmd = exec.Command("pulumi", "plugin", "install", "resource", "awsx")
+	err = cmd.Run()
+	if err != nil {
+		log.Fatalf("Command execution failed: %v", err)
+	}
+
+	// Act.
+	err = runConvert(
+		pkgWorkspace.Instance,
+		env.Global(),
+		[]string{}, /*args*/
+		cwd,        /*cwd*/
+		[]string{}, /*mappings*/
+		"pcl",      /*from*/
+		"go",       /*language*/
+		outDir,
+		true, /*generateOnly*/
+		true, /*strict*/
+		"",
+	)
+	assert.NoError(t, err)
+
+	// Assert.
+	goBytes, err := os.ReadFile(filepath.Join(outDir, "main.go"))
+	assert.NoError(t, err)
+
+	goldBytes, err := os.ReadFile("convert_aliased_imports/gold/main.go.gold")
```

Comment:
This test probably shouldn't be here in convert. It should be added to pkg/codegen/testing/test/testdata as part of the programgen tests. It's worth running it for every language to check the other codegens do correctly handle it as well.

_Even more ideally_ we'd write it in as a conformance test (cmd/pulumi-test-language) rather than using the aws schema.

Reply:
Alrighty this is done much better, thank you for letting me know )